modules/core/src/test/scala/org/tessellation/infrastructure/trust/storage/TrustStorageSuite.scala

```diff
@@ -0,0 +1,361 @@
+      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
+    } yield (TrustMap.empty.copy(trust = trustInfo), SnapshotOrdinal(ordinal))
+
+    forall(gen) {
+      case (trust, ordinal) =>
+        for {
+          store <- TrustStorage.make(trust)
+          storedTrust <- store.getTrust
+
+          maybeOrdinalPublicTrust <- store.updateNext(ordinal)
+
+          next <- store.getNextOrdinalTrust
+        } yield
+          expect.all(
+            trust == storedTrust,
+            SnapshotOrdinal(snapshotInterval) > ordinal,
+            maybeOrdinalPublicTrust.isEmpty,
+            TrustMap.empty == next.trust,
+            SnapshotOrdinal(snapshotInterval) == next.ordinal,
+            PublicTrustMap.empty == next.peerLabels
+          )
+    }
+  }
+
+  test("next ordinal trust storage updates from internal storage because expected ordinal has been reached") {
+    val minOrdinalValue = snapshotInterval + 1L
+    val maxOrdinalValue = snapshotInterval * 2L
+
+    val gen = for {
+      trustInfo <- Gen.mapOfN(4, genPeerTrustInfo)
+      delta <- Gen.chooseNum(minOrdinalValue, maxOrdinalValue)
+      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
+    } yield (TrustMap.empty.copy(trust = trustInfo), SnapshotOrdinal(ordinal))
+
+    forall(gen) {
+      case (trust, ordinal) =>
+        for {
+          store <- TrustStorage.make(trust)
+
+          maybeOrdinalPublicTrust <- store.updateNext(ordinal)
+
+          expectedOrdinalPublicTrust = trust.toPublicTrust
+          next <- store.getNextOrdinalTrust
+        } yield
+          expect.all(
+            SnapshotOrdinal(snapshotInterval) < ordinal,
+            maybeOrdinalPublicTrust.map(_.ordinal) == SnapshotOrdinal(1000L).some,
+            maybeOrdinalPublicTrust.map(_.labels) == expectedOrdinalPublicTrust.some,
+            trust == next.trust
+          )
+    }
+  }
+
+  test(
+    "current ordinal trust storage doesn't update from next ordinal trust storage because the delay expected ordinal has not been reached"
+  ) {
+    val minOrdinalValue = snapshotInterval.value
+    val maxOrdinalValue = snapshotInterval + delayToUpdateCurrent - 1L
+
+    val gen = for {
+      trustInfo <- Gen.mapOfN(4, genPeerTrustInfo)
+      delta <- Gen.chooseNum(minOrdinalValue, maxOrdinalValue)
+      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
+    } yield (TrustMap.empty.copy(trust = trustInfo), SnapshotOrdinal(ordinal))
+
+    forall(gen) {
+      case (trust, ordinal) =>
+        for {
+          store <- TrustStorage.make(trust)
+
+          _ <- store.updateNext(ordinal)
+
+          next <- store.getNextOrdinalTrust
+
+          _ <- store.updateCurrent(ordinal)
+
+          expectedCurrent <- store.getCurrentOrdinalTrust
+        } yield
+          expect.all(
+            SnapshotOrdinal(snapshotInterval |+| delayToUpdateCurrent) > ordinal,
+            trust == next.trust,
+            OrdinalTrustMap.empty == expectedCurrent
```

Comment:
aa

Reply:
All equality in the tests I created have been updated to use `===`.